pkg/middleware/requestid/requestid.go

```diff
@@ -15,10 +14,13 @@ import (
 const (
 	requestIDCtxKey   = "request-id-context-key"
 	requestIDTraceKey = "request_id"
-	RequestIDHeader   = "X-Request-Id"
+
+	// RequestIDHeader represents the HTTP header name used for
```

Comment:
This is a bit confusing as we're not passing a unique request idenfitier we're returning one.

I'd suggest

```
// RequestIDHeader defines the HTTP header that is set in each HTTP response for a given request. The value of the header is unique per request.
```